app-creator/deserializer.js

```diff
@@ -0,0 +1,438 @@
+  style.width = '100%';
+  style.height = '100%';
+
+  var map = ui.Map({ style: style });
+  var panel = ui.Panel({ style: panelStyle, widgets: [map] });
+
+  var uniqueAttributes = obj.uniqueAttributes;
+
+  // Center properties.
+  var zoom = parseInt(uniqueAttributes.zoom, 10);
+  zoom = isNaN(zoom) ? 0 : zoom;
+  var lat = parseFloat(uniqueAttributes.latitude);
+  lat = isNaN(lat) ? 0 : lat;
+  var lng = parseFloat(uniqueAttributes.longitude);
+  lng = isNaN(lng) ? 0 : lng;
+  map.setCenter(lng, lat, zoom);
+
+  // Control visibility properties.
+  var zoomControl = uniqueAttributes.zoomControl === 'true';
+  var fullscreenControl = uniqueAttributes.fullscreenControl === 'true';
+  var mapTypeControl = uniqueAttributes.mapTypeControl === 'true';
+  map.setControlVisibility({
+    drawingToolsControl: true,
+    zoomControl: zoomControl,
+    fullscreenControl: fullscreenControl,
+    mapTypeControl: mapTypeControl,
+  });
+
+  // Map styling.
+  var defaultMapStyles = uniqueAttributes.mapStyles;
+  var customMapStyles = uniqueAttributes.customMapStyles;
+  var customMapSylesJSON = null;
+  if (customMapStyles !== '') {
+    try {
+      customMapStylesJSON = JSON.parse(customMapStyles);
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  var appliedStyles =
+    customMapStyles === '' || customMapStylesJSON === null
+      ? mapStyles[defaultMapStyles]
+      : customMapStylesJSON;
+  map.setOptions({ styles: { custom: appliedStyles } });
+
+  return { node: panel, map: map };
+};
+
+/**
+ * Since Sets are not supported here, we are using an object for constant time access. The boolean values are meaningless.
+ */
+
+var unsupportedKeys = {
+  borderWidth: true,
+  borderStyle: true,
+  borderColor: true,
+  boxSizing: true,
+  'box-sizing': true,
+  top: true,
+  left: true,
+  right: true,
+  bottom: true,
+};
+
+/**
+ * Takes in a style object and returns all the unsupported keys.
+ */
+
+app.filterStyleObject = function (obj) {
+  var clone = {};
+
+  // Combine border properties.
+  var borderWidth = obj.borderWidth;
+  var borderStyle = obj.borderStyle;
+  var borderColor = obj.borderColor;
+  var border = borderWidth + ' ' + borderStyle + ' ' + borderColor;
+  clone.border = border;
+
+  if ('position' in obj && obj.position === 'absolute') {
+    clone.position = app.getPosition(obj);
+  } else {
+    unsupportedKeys.position = true;
+  }
+
+  for (var key in obj) {
+    if (!(key in unsupportedKeys)) {
+      clone[key] = obj[key];
+    }
+  }
+
+  return clone;
+};
+
+/**
+ * Create in-memory widget tree.
+ */
+app.init = function () {
+  /**
+   * Examples:
+   * - leftSidePanel
+   * - rightSidePanel
+   * - legendExample
+   * - legendTwoExample
+   * - silverExample
+   * - retroExample
+   */
+  app.widgetTreeRoot = app.deserializeUI(examples.leftSidePanel);
+};
+
+/**
+ * Draw UI to the screen by adding the widgetTreeRoot to ui.Root.
+ */
+app.draw = function () {
+  if (app.widgetTreeRoot === null) {
```

Comment:
Instead of erroring, we can call `app.init` on behalf of the user. We'll just need to document that the user should call `init` if they don't want their UI rendered immediately.